backend/backend/core/context.py

```diff
@@ -15,4 +15,4 @@ def __init__(self):
         self.models = ModelRegistry()
         self.tools = ToolRegistry()
         self.chains = ChainRegistry(self.models, self.tools)
-        self.agents = AgentRegistry(self.models, self.tools)
+        self.agents = AgentRegistry(self.models, self.tools, self.chains)
```

Comment:
The addition of `self.chains` to the AgentRegistry constructor may indicate a design flaw. Consider revisiting the dependency relationships between these classes.